### PR TITLE
chore(web): conversion of Web's browser-integration auto-tests for @web/test-runner use 🏃

### PR DESCRIPTION
**Pull request by jahorton**
Also known as web/src/test/dom, since that's its current name.

All the fixture work related to these tests makes for quite a number of changed lines, but fortunately there are pretty strong patterns throughout.

Also of note:  I moved one of the tests from the "end to end" style tests into this section, as it's all about "wrapping" specific types of context-supporting elements with our `OutputTarget` abstraction.  Now that we have a better setup for running these against all main browser engines, that set of tests fits better here.

@keymanapp-test-bot skip

### CODE REVIEW

File: web/src/test/auto/dom/cases/element-wrappers/element_interfaces.spec.ts
Comment (jahorton) [inline web/src/test/auto/dom/cases/element-wrappers/element_interfaces.spec.ts:1]: This is the test file that hops from web/src/test/auto/integrated into web/src/test/auto/dom.

File: web/test.sh [hunk 1]
```diff
@@ -21,72 +21,53 @@ builder_describe "Runs the Keyman Engine for Web unit-testing suites" \
...
-
-function get_default_browser_set() {
-  if [[ $BUILDER_OS == mac ]]; then
-    BROWSERS="--browsers Firefox,Chrome,Safari"
-  elif [[ $BUILDER_OS == win ]]; then
-    BROWSERS="--browsers Firefox,Chrome"
-  else
-    BROWSERS="--browsers Firefox,Chrome"
-  fi
-}
```
Comment (jahorton) [inline web/test.sh:60]: Source of the current build failure:

./test.sh: line 80: get_default_browser_set: command not found

Looks like I rebased this deletion a bit too early, so the CI check falls over here.  It's fine as of the next PR in the chain though - #11456 passed its checks.

**Review by jahorton — COMMENTED** [review]
Note: /web/build.sh was not adjusted to include `install playwright` as part of its configure.

Reasons:
1. It currently doesn't have its own `configure` action - it has child projects for that.
2. It and its child projects (`engine/osk`) have dependencies on projects that _do_ have `configure` actions that'll run it anyway.

**Review by mcdurdin — APPROVED** [review]

**Review by ermshiperete — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.42-alpha](https://keyman.com/downloads/releases/alpha/18.0.42)

**Pull request merged**